lib/ansible/modules/user.py

```diff
@@ -868,8 +873,10 @@ def modify_user_usermod(self):
                                     cmd.append('-a')
                                 groups_need_mod = True
                                 break
+                        self.groups_diff['after'] = list(set(current_groups) | groups)
```

Comment:
Can we apply `list()` outside if/else to reduce complexity? It's already the 5th level of nesting, let's not increase the inline nesting additionally to that…

Reply:
There's a mix of lists and sets, and I fixed the excessive nesting by casting the lists to sets instead. This is more fitting anyway, since the order doesn't matter internally.